src/lib/select/select.ts

```diff
@@ -1119,6 +1122,11 @@ export class MatSelect extends _MatSelectMixinBase implements AfterContentInit,
     const maxOptionsDisplayed = Math.floor(SELECT_PANEL_MAX_HEIGHT / itemHeight);
     let optionOffsetFromPanelTop: number;
 
+    // Disable offset for smaller lists by returning 0 as value to offset
+    if (this.disableOffset == 'true') {
+      return 0;
```

Comment:
The new option still needs unit tests.

Reply:
I've added a unit test `with option centering disabled` along with a component for testing it. I was thinking of adding it to the 'positioning' tests, but didn't do so to make it easier to add other cases. 
I've run `gulp test` after adding it and the headless chrome returns all successful, but I'm not sure if there is a better way to test it.